src/components/icon.js

```diff
@@ -5,20 +5,78 @@
   orientation: 'VERTICAL',
   jsx: (() => {
     const { Icons } = window.MaterialUI;
+    const { Link, Badge } = window.MaterialUI.Core;
+    const { useText } = B;
     const isDev = B.env === 'dev';
-    const { icon } = options;
+    const {
+      icon,
+      addBadge,
+      content,
+      badgeColor,
+      invisible,
+      anchorOrigin,
+      variant,
+      linkTo,
+      linkToExternal,
+      linkType,
+    } = options;
+
+    const hasLink = linkTo && linkTo.id !== '';
+    const hasExternalLink = linkToExternal && linkToExternal.id !== '';
+
+    const Content = useText(content);
+    const anchorOriginSplit = anchorOrigin.split(',');
+    const anchorOriginObj = {
+      horizontal: anchorOriginSplit[0],
+      vertical: anchorOriginSplit[1],
+    };
+
     const IconComponent = React.createElement(Icons[icon], {
       className: classes.root,
     });
 
+    const LinkComponent = (
+      <Link
+        href={
+          linkType === 'external' && hasExternalLink
+            ? linkToExternal
+            : 'https://google.com'
+        }
+        component={linkType === 'internal' && hasLink ? B.Link : undefined}
+        endpoint={linkType === 'internal' && hasLink ? linkTo : undefined}
+      >
+        {IconComponent}
+      </Link>
+    );
+
+    const Icon = hasLink ? LinkComponent : IconComponent;
+
+    const BadgeComponent = (
+      <div className={classes.badge}>
+        <Badge
+          badgeContent={Content}
+          color={badgeColor}
+          invisible={invisible}
+          anchorOrigin={anchorOriginObj}
+          variant={variant}
```

Comment:
When this variant equals `dot` I believe you will need this property as well, to position it better: `overlap="circle"` https://material-ui.com/components/badges/#badge-overlap

Reply:
Fixed